core/src/main/java/org/apache/calcite/materialize/Lattice.java

```diff
@@ -513,11 +513,11 @@ public boolean isAlwaysMeasure(Column column) {
     }
 
     Vertex getTarget() {
-      return (Vertex) target;
+      return target;
```

Comment:
Can we just make source() and target() base class methods? So these can be removed.

Reply:
Revised accordingly. Thanks for the good suggestion.